pkg/controllers/provisioning/provisioner.go

```diff
@@ -111,11 +111,20 @@ func (p *Provisioner) provision(ctx context.Context) error {
 	logging.FromContext(ctx).Infof("Batched %d pod(s) in %s", len(pods), window)
 
 	// Schedule pods to potential nodes
-	nodes, err := p.schedule(ctx, pods)
+	nodes, inflight, err := p.schedule(ctx, pods)
 	if err != nil {
 		return err
 	}
 
+	// TODO(todd): do we want to do this?  There may be taints that pods don't tolerate.
+	// Bind pods to in-flight nodes
+	workqueue.ParallelizeUntil(ctx, len(inflight), len(inflight), func(i int) {
+		node := inflight[i]
```

Comment:
If we do do this, should we have a debug log describing a bind to an in-flight node? It might surprise users that are expecting a new node to launch but none were

Reply:
I'm about to push up a new PR that starts to add support for recording events on the pods (at least an interface for it).  We needed something like that for unit tests anyway as we need to know if the scheduling decisions were correct, even without the pod binding.  There is a separate issue at #1584 to turn those into actual events on the pods.  It's more code and requires changing permissions, so I didn't include it in this PR.